infor-design/enterprise
# 440 - Respect `groupSize` of defined locale in Mask components

**Explain the _details_ for making this change. What existing problem does the pull request solve?**
This PR enables the passing of different locales into the Mask, to more easily take advantage of the locale's preferences for formatting numbers.  This fixes a problem where some locales with different `groupSize` definitions for thousands separators were not being respected.

Additionally, this PR includes a change to the `ar-EG` locale's thousands separator, making it a decimal instead of the original comma.  This makes the keyboard output on Egyptian Arabic keyboards match what's expected by the Mask for proper formatting (previously on this locale, it was not possible to properly format the Mask fields with decimals)

**Related github/jira issue (required)**:
Closes #440 

**Steps necessary to review your pull request (required)**:
Pull this branch and run the demoapp, then:

_Test the groupSize setting:_
- Open http://localhost:4000/components/mask/test-number-mask-gauntlet.html
- Test fields that include thousands separators and decimals to ensure formatting while typing is working correctly.  Numbers should have the standard group size ([3, 0])
- Open http://localhost:4000/components/mask/test-number-mask-gauntlet.html?locale=nl-NL
- Test the "Way Longer Number" fields with decimals/thousands to make sure they are working correctly (groupSize is [3,3])
- Open http://localhost:4000/components/mask/test-number-mask-gauntlet.html?locale=hi-IN
- Test the "Way Longer Number" fields with decimals/thousands to make sure they are working correctly (groupSize is [3,2])

_Test the decimal change on `ar-EG`:_
- Open http://localhost:4000/components/mask/test-number-mask-gauntlet.html?locale=ar-EG
- Type numbers into fields that handle decimal points, then type the decimal point.  It should correctly propagate.

Additionally, all e2e/functional tests should pass.

----
paging @fitzorama
